test/cpp/microbenchmarks/bm_event_engine_run.cc

```diff
@@ -0,0 +1,220 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <atomic>
+#include <cmath>
+#include <memory>
+
+#include <benchmark/benchmark.h>
+
+#include <grpc/event_engine/event_engine.h>
+#include <grpcpp/impl/grpc_library.h>
+
+#include "src/core/lib/event_engine/common_closures.h"
+#include "src/core/lib/event_engine/default_event_engine.h"
+#include "src/core/lib/event_engine/promise.h"
+#include "test/core/util/test_config.h"
+#include "test/cpp/microbenchmarks/helpers.h"
+#include "test/cpp/util/test_config.h"
+
+namespace {
+
+using ::grpc_event_engine::experimental::AnyInvocableClosure;
+using ::grpc_event_engine::experimental::EventEngine;
+using ::grpc_event_engine::experimental::GetDefaultEventEngine;
+using ::grpc_event_engine::experimental::Promise;
+using ::grpc_event_engine::experimental::ResetDefaultEventEngine;
+
+void BM_EventEngine_RunLambda(benchmark::State& state) {
+  int cb_count = state.range(0);
+  std::atomic_int cnt{0};
+  Promise<bool> p{false};
+  auto cb = [&cnt, &p, cb_count]() {
+    if (++cnt == cb_count) p.Set(true);
+  };
+  auto engine = GetDefaultEventEngine();
+  for (auto _ : state) {
+    for (int i = 0; i < cb_count; i++) {
+      engine->Run(cb);
+    }
+    GPR_ASSERT(p.Get());
```

Comment:
I think we should remove the GPR_ASSERTS to avoid distorting the timing info

Reply:
Done. It's safe enough since it's a boolean that's only set in one way. I left in one `GPR_DEBUG_ASSERT`, since benchmarks shouldn't normally be run in debug mode, and it might help catch things in development.